### PR TITLE
feat(downloader): improved notifier experience

### PR DESCRIPTION
**Pull request by ithinkicancode**

### CODE REVIEW

**Comment by shanewilson** [comment]
👍 

does @phuongmy like it?

**Comment by phuongmy** [comment]
Yes it looks good. But I would like to play with it on Dev before we merge it.

**Comment by phuongmy** [comment]
Note: this PR is for PRTL-44

File: app/scripts/components/downloader/downloader.directive.ts [hunk 1]
```diff
@@ -121,14 +135,15 @@ module ngApp.components.downloader.directive {
...
           if (++attempts % timeoutInterval === 0) {
-            notify.closeAll();
-            notify({
-              message: null,
-              messageTemplate: (altMessage && attempts > timeoutInterval * 2) ? detailedMessage : simpleMessage,
-              container: '#notification',
-              classes: 'alert-warning',
-              scope: notifyScope
-            });
+            if (! alreadyShowingAltMessage && shouldShowAltMessage() && ! noToastMessageInDisplay()) {
```
Comment (cy) [inline app/scripts/components/downloader/downloader.directive.ts]: space after `!`s?

File: app/scripts/components/downloader/downloader.directive.ts [hunk 2]
```diff
@@ -85,6 +84,21 @@ module ngApp.components.downloader.directive {
         finished();
       };
 
+      const shouldShowAltMessage = () => altMessage && attempts > timeoutInterval * 2;
+      const notifierDomHost = '#notification';
+      const noToastMessageInDisplay = () => $(notifierDomHost).children().length == 0;
```
Comment (cy) [inline app/scripts/components/downloader/downloader.directive.ts]: ```===```

File: app/scripts/components/downloader/downloader.directive.ts (hunk 1 above)
Comment (alex-wilmer) [inline app/scripts/components/downloader/downloader.directive.ts]: I think I've seen that somewhere but I can't remember anymore. I'll ask Terry next time I see him :)

**Pull request merged**